### PR TITLE
[mds-*] Massive refactoring to comply with async/await linter rules

### PR DESCRIPTION
**Pull request by avatarneil**
Finished converting so far:
* mds-agency
* mds-db
* mds-logger
* mds-daily
* mds-cache

## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [x] Provider
- [x] Agency
- [x] Audit
- [x] Policy
- [x] Compliance

### CODE REVIEW

File: packages/mds-cache/index.ts [hunk 1]
```diff
@@ -124,16 +115,12 @@ async function hread(suffix: string, device_id: UUID): Promise<CachedItem> {
...
-        // log.info(`hread: ${suffix} for ${device_id} not found`)
-        reject(new Error(`${suffix} for ${device_id} not found`))
-      }
-    })
-  })
+  const flat = await (await getClient()).hgetallAsync(key)
+  if (flat) {
+    return unflatten({ ...flat, device_id })
+  }
+  // log.info(`hread: ${suffix} for ${device_id} not found`)
```
Comment (janedotx) [inline packages/mds-cache/index.ts]: Let's just delete this commented out line.

File: packages/mds-cache/index.ts [hunk 2]
```diff
@@ -148,7 +135,11 @@ async function hreads(suffixes: string[], device_ids: UUID[]): Promise<CachedIte
 
   suffixes.map(suffix => device_ids.map(device_id => multi.hgetall(`device:${device_id}:${suffix}`)))
 
+  /* eslint-reason external lib weirdness */
```
Comment (janedotx) [inline packages/mds-cache/index.ts]: Which lib?

File: packages/mds-cache/index.ts [hunk 3]
```diff
@@ -185,65 +176,60 @@ async function readDevices(device_ids: UUID[]) {
...
+  // big batch redis nightmare!
+  let all: CachedItem[] = await hreads(['device', 'event', 'telemetry'], device_ids)
+  all = all.filter((item: CachedItem) => Boolean(item))
+  all.map(item => {
+    device_status_map[item.device_id] = device_status_map[item.device_id] || {}
+    Object.assign(device_status_map[item.device_id], item)
   })
+  log.info('readDevicesStatus', device_ids.length, 'entries:', all.length)
+
+  // log.info('device_status_map', JSON.stringify(device_status_map))
```
Comment (janedotx) [inline packages/mds-cache/index.ts]: let's kill this too

File: packages/mds-cache/index.ts [hunk 4]
```diff
@@ -490,32 +461,28 @@ async function cleanup(deviceIdMap: { [device_id: string]: boolean }) {
...
-      }
-      resolve(report)
-    })
+    try {
+      // look for bogus keys
+      let badKeys: string[] = []
+      keys.map(key => {
+        const [, device_id, suffix] = key.split(':')
+        if (deviceIdMap[device_id]) {
+          // woot
```
Comment (janedotx) [inline packages/mds-cache/index.ts]: remove this

File: aws-lambda/.eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (mdurling) [inline aws-lambda/.eslintrc.js]: Why do we need another `eslintrc` file?

File: packages/.eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
```
Comment (mdurling) [inline packages/.eslintrc.js]: Why do we need this file?

File: tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (mdurling) [inline tsconfig.json]: Why are we adding a root `tsconfig`? All of the packages extend the `tsconfig.settings` file that has all of these values? I am not opposed to adding all of these extra files, but I would like to understand what improvement this provides. It has been an ongoing challenge getting lerna/yarn/typescript/eslint/vscode/webpack all working together.

Comment (avatarneil) [reply tsconfig.json]: I had to add all of these to get projectOptions working properly with `no-floating-promises`. I tried to find a way to get everything working without them, but no dice 😢

File: packages/mds-cache/index.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-cache/index.ts]: redis

File: aws-lambda/.eslintrc.js (hunk 5 above)
Comment (avatarneil) [reply aws-lambda/.eslintrc.js]: Necessary for parserOptions.

File: packages/.eslintrc.js (hunk 6 above)
Comment (avatarneil) [reply packages/.eslintrc.js]: Moar parserOptions. Remove it and things go sads.

Comment (mdurling) [reply packages/.eslintrc.js]: Ok. I have another approach I can try once the dust settles.

File: aws-lambda/aws-utils/index.ts [hunk 8]
```diff
@@ -5,20 +5,18 @@ import KMS from 'aws-sdk/clients/kms'
...
-    })
-  })
+    if (!data || !data.Plaintext) {
+      throw new Error('no response to decryption attempt')
+    } else {
+      return data.Plaintext.toString('ascii')
+    }
+  } catch (err) {
+    throw err
+  }
```
Comment (karcass) [inline aws-lambda/aws-utils/index.ts]: should we be closing the file handle?

File: packages/mds-policy/package.json [hunk 9]
```diff
@@ -29,8 +29,8 @@
     "test": "yarn test:prettier && yarn test:eslint && yarn test:unit",
     "test:eslint": "eslint --ignore-path ../../.gitignore '**/*.ts'",
     "test:prettier": "prettier --write --check --ignore-path ../../.gitignore '**/*.ts'",
-    "test:unit": "nyc --check-coverage --exclude tests --extension .ts --lines 80 --reporter=text --reporter=html ts-mocha --require source-map-support/register --recursive tests/**/*.ts",
+    "test:unit": "nyc --check-coverage --exclude tests --extension .ts --lines 79 --reporter=text --reporter=html ts-mocha --require source-map-support/register --recursive tests/**/*.ts",
```
Comment (janedotx) [inline packages/mds-policy/package.json]: y

Comment (avatarneil) [reply packages/mds-policy/package.json]: Added some moar tests to improve coverage back to its previous glory!

**Pull request merged**